src/process_utility.c

```diff
@@ -365,8 +365,22 @@ process_copy(ProcessUtilityArgs *args)
 	Cache *hcache;
 	Oid relid;
 
+	/* We only copy for COPY FROM. Since hypertables do not contain any data,
+	 * we skip the copy for COPY TO, but print an informative message. */
 	if (!stmt->is_from || NULL == stmt->relation)
+	{
+		if (stmt->relation)
+			ereport(NOTICE,
+					(errmsg("hypertables has no data in main table, no data will be copied"),
```

Comment:
Suggestion:
> COPY TO copies no data for hypertable

Reply:
That is strictly not true, `COPY (SELECT * FROM hyper) TO STDOUT` does copy data.

Reply:
> That is strictly not true, `COPY (SELECT * FROM hyper) TO STDOUT` does copy data.

In your example, COPY TO is applied to select statement, not to hypertable directly. If you think my formulation is not good enough (and I said I don't think it is ideal), you are welcome to come up with better message, but the original formulation is no go, imho.

Reply:
Yes, but users are probably not aware of such fine points. I'll figure out some other formulation that does not talk about main and child tables and uses `COPY TO`.